### ISSUE TITLE
Add a `migration_mode` flag to the `console` CLI options

### ISSUE THREAD

**Issue by KCErb** [hasura/graphql-engine#2290]
When using the cli `console` to modify your metadata, you can turn off migration_mode with a little toggle switch. This is great for when you are managing your own schema and don't want hasura to touch anything except its own metadata.

The trouble is that this little toggle is stateful, so if you're running out of docker, you have to remember to flip that switch each time you spin up a container.

It'd be great to just have a flag you can pass:

```
hasura console --migration-mode false
```

or

```
hasura console --no-migrations
```

The work around at the moment is to either:
* use curl (or similar) to mimic the API call that does this:

```
curl -H 'Content-Type: application/json' -X PUT -d '{name: "migration_mode", value: "false"}' http://0.0.0.0:9693/apis/migrate/settings
```
but that actually doesn't work because cookies guard against such a thing, so either use your favorite "stores cookies in memory" library and script it, or use curl and write them to a file

* execute some SQL when you spin up your postgres container:

```
CREATE TABLE IF NOT EXISTS hdb_catalog.migration_settings (setting text not null primary key, value text not null);

INSERT INTO hdb_catalog.migration_settings (setting, value) VALUES ('migration_mode', 'false') ON CONFLICT ON CONSTRAINT migration_settings_pkey 

DO NOTHING;
```

(Thank you to the awesome support team, and especially @shahidhk for the required SQL.)

**Comment by shahidhk** [comment]
The best way to do this would be to have a command dedicated to doing this, instead of a flag for console command.

A flag for console command indicates that this setting is for the session of the console, but this setting is persisted on the database and is available to all consoles connected to this database.

The command could look like this:
```
hasura settings --get <key>
hasura settings --set <key>=<value>
```

Suggestions/comments are invited.

#### Implementation
The plumbing required to do this already written. The function is [`UpdateSetting`](https://github.com/hasura/graphql-engine/blob/master/cli/migrate/database/settings.go#L20) and it is implemented in the [`hasuradb` driver](https://github.com/hasura/graphql-engine/blob/master/cli/migrate/database/hasuradb/settings.go#L175).

**Comment by KCErb** [comment]
Awesome, I'll see if I can't put together a PR for this this week.

**Comment by KCErb** [comment]
I think I've got a good start here, but not sure how to get the right parameters [right here](https://github.com/KCErb/graphql-engine/blob/5feef8443df23eafbc17ec7413b138be5f4d4eaf/cli/commands/settings.go#L49-L50).

Also not sure what kind of tests could be written.

If you uncomment the given lines you should get a null pointer error from somewhere. I don't know if it's those URLs or something else so please let me know what I can do to take this the next step towards a PR.

**Comment by shahidhk** [comment]
@arvi3411301 Can you checkout @KCErb's comment above?

**Comment by arvi3411301** [comment]
@KCErb you need to pass both databaseURL and sourceURL as arguments to migrate.New method.

**Comment by KCErb** [comment]
@arvi3411301 right, how can I get that information into this context? Is it somehow available on the ec like I have in the commented out part? Or do I need to get it from somewhere else? The migration command gets it from the gin context but I don't think I have that available right?

**Comment by arvi3411301** [comment]
@KCErb you can refer migration status command [here](https://github.com/hasura/graphql-engine/blob/master/cli/commands/migrate_status.go).

**Comment by arvi3411301** [comment]
@shahidhk Can we move settings command as a sub command of hasura migration?

**Comment by KCErb** [comment]
@arvi3411301 oh I see, I can use that `newMigrate` function to construct from the EC, nice! Thanks

@shahidhk  In the spirit of "can we move" I was also wondering about doing this as a sub-command structure:

```
hasura settings get xxx
hasura settings set xxx=yyy
```

if we do it as flags, that seems to imply that we could do both:

```
hasura settings --set migration_mode=false --get migration_mode
```

but the expected / desired output of a such a thing might not always be so easy to determine. So a sub command approach might help make it clear that we are just performing one operation.

**Comment by shahidhk** [comment]
@KCErb We can do 

`hasura migrate settings set <key=value>` and
`hasura migrate settings get <key>`

So that you re-use the code from `migrate` package as @arvi3411301 mentioned.

**Comment by KCErb** [comment]
Hi @marionschleifer I noticed you added the help wanted label to this issue. What help is needed? Last I heard on the PR (2321) this was waiting on internal review. Is that wrong?

**Comment by marionschleifer** [comment]
@KCErb you are totally right. I missed this. Removing it. Thanks for letting me know 🙂
